### ISSUE TITLE
Not able to search in relations of relations (Nested Relations throw missing FROM-clause entry)

### ISSUE THREAD

**Issue by Alfagun74** [ppetzold/nestjs-paginate#405]
I wanted to implement a search using a property from a relation of a relation from my Entity.

``json
searchableColumns: [
        "title",
        "games.genres.name"
      ],
``

But i get this:
```log
error:   ┏ missing FROM-clause entry for table "genres"
error:   ┃ [ 1] QueryFailedError: missing FROM-clause entry for table "genres"
error:   ┃ [ 2]     at PostgresQueryRunner.query (/app/node_modules/.pnpm/typeorm@0.3.10_pg@8.8.0+ts-node@10.9.1/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:211:19)
error:   ┃ [ 3]     at runMicrotasks (<anonymous>)
error:   ┃ [ 4]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
error:   ┃ [ 5]     at async SelectQueryBuilder.loadRawResults (/app/node_modules/.pnpm/typeorm@0.3.10_pg@8.8.0+ts-node@10.9.1/node_modules/typeorm/query-builder/SelectQueryBuilder.js:2056:25)
error:   ┃ [ 6]     at async SelectQueryBuilder.getRawMany (/app/node_modules/.pnpm/typeorm@0.3.10_pg@8.8.0+ts-node@10.9.1/node_modules/typeorm/query-builder/SelectQueryBuilder.js:607:29)
error:   ┃ [ 7]     at async SelectQueryBuilder.executeEntitiesAndRawResults (/app/node_modules/.pnpm/typeorm@0.3.10_pg@8.8.0+ts-node@10.9.1/node_modules/typeorm/query-builder/SelectQueryBuilder.js:1860:26)
error:   ┃ [ 8]     at async SelectQueryBuilder.getManyAndCount (/app/node_modules/.pnpm/typeorm@0.3.10_pg@8.8.0+ts-node@10.9.1/node_modules/typeorm/query-builder/SelectQueryBuilder.js:771:36)
error:   ┃ [ 9]     at async paginate (/app/node_modules/.pnpm/nestjs-paginate@4.5.1_5lrjep46ocnlg5z2ezquymw4bm/node_modules/nestjs-paginate/lib/paginate.js:283:27)
error:   ┃ [10]     at async TagsController.get (/app/dist/controllers/tags.controller.js:32:34)
error:   ┃ [11]     at async /app/node_modules/.pnpm/@nestjs+core@9.2.0_jfowmhmgk7kwjudosrpv7e7liy/node_modules/@nestjs/core/router/router-execution-context.js:46:28
error:   ┗ [12] { context: 'ExceptionsHandler' }
```

Aren't nested relations supported in the search?

**Comment by ppetzold** [comment]
have you added `relations` prop to left join the table?

https://github.com/ppetzold/nestjs-paginate/blob/master/src/paginate.spec.ts#L854

should be working as test shows.

**Comment by ppetzold** [comment]
happy to accept PR with bug fix if there are any issues

**Comment by dhruv1409** [comment]
`@ManyToOne(() => GeographyClassificationEntity, geography => geography.id, {
    eager: true,
  })
  @JoinColumn({ name: 'geographyClassification', referencedColumnName: 'id' })
  @Column({ name: 'geographyClassification', nullable: true })
  public geographyClassification: number;

  @OneToMany(
    () => ComponentTemplateAllowedComponentTypeMappingEntity,
    mappingTable => mappingTable.componentTemplate,
    { eager: true },
  )
  public allowedComponentTypes: ComponentTemplateAllowedComponentTypeMappingEntity[];
`
I have these 2 relations set in my entity . So if I try like this - 

` const result = await paginate(query, this.lcaProcessRepo, {
        sortableColumns: ['id', 'isGeneralOwnership'],
        searchableColumns: [, 'description'],
        defaultSortBy: [['id', 'DESC']],
        defaultLimit: 10,
        maxLimit: 50,
        filterableColumns: {
          geographyClassification: [FilterOperator.IN],
          processTechnology: [FilterOperator.IN],
        },
        relations: ['geographyClassification', 'allowedComponentTypes'],
  });`

It only accepts allowedComponentTypes (which is ManyToOne). 

If anyone can fix this issue - or let me know if I am missing anything here ..

**Comment by Alfagun74** [comment]
> have you added `relations` prop to left join the table?
> 
> https://github.com/ppetzold/nestjs-paginate/blob/master/src/paginate.spec.ts#L854
> 
> should be working as test shows.

i think that was it for me thank you

**Comment by Helveg** [comment]
@dhruv1409 your `ManyToOne` decoration seems to contain a lot of unconventional specifications that TypeORM determines automatically. Could you explain why you chose your decorators over this conventional TypeORM pattern:

```
@ManyToOne(() => GeographyClassificationEntity, {eager: true})
public geographyClassification: GeographyClassificationEntity
```

nestjs-paginate supports every type of TypeORM relationship.

**Comment by Helveg** [comment]
@ppetzold I noticed that the tests never specify nested relations in the `relations` key, and that when I try it, I get an error. Passing the same relations to `repo.find` behaves as expected:

```
const pagination = this.getPaginationConfig();
console.log(pagination);
await this.service.repo.find({ relations: pagination.relations });
console.log('We can find it through repo!');
return await paginate(
  <PaginateQuery>query,
  this.service.repo,
  pagination,
)
```

Error log:

```
{
  sortableColumns: [ 'createdAt' ],
  filterableColumns: { 'campaign.id': [ '$eq' ] },
  searchableColumns: [],
  defaultLimit: 20,
  defaultSortBy: [ [ 'createdAt', 'DESC' ] ],
  relations: [ 'product', 'campaign', 'campaign.organization' ],
  where: undefined
}
We can find it through repo!
[Nest] 3512  - 01/21/2023, 12:57:34 PM   ERROR [ExceptionsHandler] Relation with property path campaign.organization in entity was not found.
TypeORMError: Relation with property path campaign.organization in entity was not found.
    at JoinAttribute.getValue (C:\Users\pwd06\git\graag-gedaan\src\query-builder\JoinAttribute.ts:180:23)
    at JoinAttribute.get relation [as relation] (C:\Users\pwd06\git\graag-gedaan\src\query-builder\JoinAttribute.ts:184:53)
    at JoinAttribute.get metadata [as metadata] (C:\Users\pwd06\git\graag-gedaan\src\query-builder\JoinAttribute.ts:196:18)
    at SelectQueryBuilder.join (C:\Users\pwd06\git\graag-gedaan\src\query-builder\SelectQueryBuilder.ts:2010:27)
    at SelectQueryBuilder.leftJoin (C:\Users\pwd06\git\graag-gedaan\src\query-builder\SelectQueryBuilder.ts:465:14)
    at SelectQueryBuilder.leftJoinAndSelect (C:\Users\pwd06\git\graag-gedaan\src\query-builder\SelectQueryBuilder.ts:605:14)
    at C:\Users\pwd06\git\graag-gedaan\node_modules\nestjs-paginate\src\paginate.ts:259:26
    at Array.forEach (<anonymous>)
    at paginate (C:\Users\pwd06\git\graag-gedaan\node_modules\nestjs-paginate\src\paginate.ts:258:26)
    at ProductOffersController.findAll (C:\Users\pwd06\git\graag-gedaan\src\core\resources\resource.controller.ts:171:30)
```

**Comment by Helveg** [comment]
Piling on to the `relations` issues, also passing a relationship twice yields a cryptic error while `repo.find` doesn't complain:

```
{
  sortableColumns: [ 'createdAt' ],
  filterableColumns: { 'campaign.id': [ '$eq' ] },
  searchableColumns: [],
  defaultLimit: 20,
  defaultSortBy: [ [ 'createdAt', 'DESC' ] ],
  relations: [ 'product', 'campaign', 'campaign' ],
  where: undefined
}
We can find it
[Nest] 20260  - 01/21/2023, 2:08:55 PM   ERROR [ExceptionsHandler] SQLITE_ERROR: ambiguous column name: e_campaign.id
QueryFailedError: SQLITE_ERROR: ambiguous column name: e_campaign.id
    at handler (C:\Users\pwd06\git\graag-gedaan\src\driver\sqlite\SqliteQueryRunner.ts:113:26)
    at replacement (C:\Users\pwd06\git\graag-gedaan\node_modules\sqlite3\lib\trace.js:25:27)
    at Statement.errBack (C:\Users\pwd06\git\graag-gedaan\node_modules\sqlite3\lib\sqlite3.js:15:21)
```

**Comment by ppetzold** [comment]
Closed by #499
